arch/power/adler32_vmx.c

```diff
@@ -0,0 +1,123 @@
+/* adler32_vmx.c -- compute the Adler-32 checksum of a data stream
+ * Copyright (C) 1995-2011 Mark Adler
+ * Copyright (C) 2017-2021 Mika T. Lindqvist <postmaster@raasu.org>
+ * For conditions of distribution and use, see copyright notice in zlib.h
+ */
+
+#ifdef PPC_VMX_ADLER32
+#include <altivec.h>
+#ifdef ZLIB_COMPAT
```

Comment:
Do we need to use these zlib.h/zlib-ng.h includes?

Reply:
As long as `zutil.h` is included first, it doesn't seem to be absolutely necessary as `zutil.h` has the logic too...